ESP/src/main.cpp

```diff
@@ -38,10 +39,34 @@ StreamServer streamServer;
 #endif  // SIM_ENABLED
 
 void etvr_eye_tracker_web_init() {
+  // Check if mode has been changed to USB mode before starting network initialization
+  DeviceModeManager* deviceModeManager = DeviceModeManager::getInstance();
+  if (deviceModeManager && deviceModeManager->getMode() == DeviceMode::USB_MODE) {
+    log_i("[SETUP]: Mode changed to USB before network initialization, aborting");
+    WiFi.disconnect(true);
+    return;
+  }
```

Comment:
We're doing the same thing three times, I'd move it out to a smaller helper function, a bit easier to maintain later